onnxruntime/core/providers/cuda/tensor/pad.cc

```diff
@@ -120,6 +120,16 @@ Status Pad<T>::ComputeInternal(OpKernelContext* ctx) const {
   }
 
   auto& output_tensor = *ctx->Output(0, output_shape);
+  if (std::all_of(p_pads->begin(), p_pads->end(), [](const int64_t v) { return v == 0; }) &&
```

Comment:
1. add a comment here to describe the purpose.
2. Can this move up to line 97? If there is no padding, then output_shape size is input size, those should be checked at very eary stage.